### PR TITLE
pythonPackages.mnist: init at 0.2.2

### PR DESCRIPTION
**Pull request by dmrauh**
###### Motivation for this change
Add python mnist utilities.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.mnist python3.pkgs.mnist

File: pkgs/development/python-modules/mnist/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    inherit pname version;
+    sha256 = "0m3vz9z7gay8nndid2784j95fzif502shfvik3f8mggvx6nlz4r4";
+  };
+
+  propagatedBuildInputs = [
+    numpy
+  ];
+
+  meta = with lib; {
+   description = "MNIST dataset";
```
Comment (dotlambda) [inline pkgs/development/python-modules/mnist/default.nix]:
```suggestion
   description = "Python utilities to download and parse the MNIST dataset";
```
no longDescription needed

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please add yourself to `maintainer-list.nix` in a separate commit and change the commit message of the existing one to the PR's title.

> Ran 0 tests in 0.000s

Please provide an appropriate `checkPhase` or, if there are no tests, set `doCheck = false` and add a comment with the reason.

Finally, please make the python2 build work.

File: pkgs/development/python-modules/mnist/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  pname = "mnist";
+  version = "0.2.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0m3vz9z7gay8nndid2784j95fzif502shfvik3f8mggvx6nlz4r4";
+  };
+
+  propagatedBuildInputs = [
+  numpy
```
Comment (dotlambda) [inline pkgs/development/python-modules/mnist/default.nix]:
```suggestion
    numpy
```

File: pkgs/development/python-modules/mnist/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  ] ++ lib.optionals (isPy27) [ mock ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/mnist/default.nix]:
```suggestion
  ] ++ lib.optional isPy27 mock;
```
I'm pretty sure that should only be a test dependency. You should make upstream aware of that.

File: pkgs/development/python-modules/mnist/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  # There exist no tests
```
Comment (dotlambda) [inline pkgs/development/python-modules/mnist/default.nix]: There are tests: https://github.com/datapythonista/mnist/tree/mnist-0.2.2/tests. However, they're probably not distributed on PyPI.

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by aanderse** [comment]
@dmrauh are you able to address the comments above and continue with this PR?

**Comment by dmrauh** [comment]
Please excuse my long hiatus. I hope that i have addressed all comments. The only problem i had was that some tests using sockets during check phase were failing which i had to disable in the end.

**Comment by dmrauh** [comment]
@dotlambda @aanderse do you think my PR can be merged or is there anything else to do?

**Comment by aanderse** [comment]
@dmrauh I don't see any issues. Thanks for following up. @dotlambda good to go from your perspective?

File: pkgs/top-level/python-packages.nix [hunk 5]
```diff
@@ -770,7 +770,9 @@ in {
 
   mkl-service = callPackage ../development/python-modules/mkl-service { };
 
-  monkeyhex = callPackage ../development/python-modules/monkeyhex { };
+  mnist = callPackage ../development/python-modules/mnist { };
+
+  monkeyhex = callPackage ../development/python-modules/monkeyhex { };
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix]: I feel like monkeyhex wasn't meant to be added, please remove

Comment (dmrauh) [reply pkgs/top-level/python-packages.nix]: I accidentally added a space after monkeyhex, hence it was added. I fixed it now.

File: pkgs/development/python-modules/mnist/default.nix [hunk 6]
```diff
@@ -0,0 +1,36 @@
+{ buildPythonPackage, fetchFromGitHub, isPy27, lib, mock, numpy, pytest
```
Comment (jonringer) [inline pkgs/development/python-modules/mnist/default.nix]: pytestCheckHook will bring pytest into scope
```suggestion
{ buildPythonPackage, fetchFromGitHub, isPy27, lib, mock, numpy

File: pkgs/development/python-modules/mnist/default.nix [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  src = fetchFromGitHub {
+    owner = "datapythonista";
+    repo = "mnist";
+    rev = "${pname}-${version}";
+    sha256 = "17r37pbxiv5dw857bmg990x836gq6sgww069w3q5jjg9m3xdm7dh";
+  };
+
+  propagatedBuildInputs = [ numpy ] ++ lib.optional isPy27 mock;
+
+  checkInputs = [ pytest pytestCheckHook ];
```
Comment (jonringer) [inline pkgs/development/python-modules/mnist/default.nix]:
```suggestion
  checkInputs = [ pytestCheckHook ];
```

Comment (dmrauh) [reply pkgs/development/python-modules/mnist/default.nix]: Thanks for the hints! I removed pytest accordingly.

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
commits LGTM

```
[3 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/56518
3 package were built:
python27Packages.mnist python37Packages.mnist python38Packages.mnist
```

**Comment by dmrauh** [comment]
Thanks to all reviewers for your help!

**Comment by jonringer** [comment]
... it  did take 8 months... that's quite a while. what's impressive to me, is that this means we've had 17k PRs in 8 months, which is averaging like 2.1k a month O.o

**Pull request merged**